src/Proxy.cpp

```diff
@@ -182,13 +182,14 @@ void Proxy::registerSignalHandlers(sdbus::internal::IConnection& connection)
         for (auto& signalItem : signalsOnInterface)
         {
             const auto& signalName = signalItem.first;
-            auto& slot = signalItem.second.slot_;
-            slot = connection.registerSignalHandler( destination_
+            auto* signalData = signalItem.second.get();
+            auto& slot = signalData->slot_;
+            slot = connection.registerSignalHandler(destination_
```

Comment:
Minor: we can now assign to `signalData->slot`. Previously, a `slot` reference was made for higher readability, to avoid mingling with non-expressive `first` and `second` members of the map value pair. In this view, now we don't need the reference :)